header.go

```diff
@@ -57,6 +57,46 @@ func mergeHeaders(into, from http.Header) {
 	}
 }
 
+// mergeMetdataHeaders merges the metadata headers from the "from" header into
+// the "into" header. It skips over non metadata headers that should not be
+// propagated from the server to the client.
+func mergeMetadataHeaders(into, from http.Header) {
+	for key, vals := range from {
+		if len(vals) == 0 {
+			// For response trailers, net/http will pre-populate entries
+			// with nil values based on the "Trailer" header. But if there
+			// are no actual values for those keys, we skip them.
+			continue
+		}
+		switch http.CanonicalHeaderKey(key) {
```

Comment:
Is the key already canonicalized at this point? It isn't being canonicalized in the other `mergeHeaders` which would also cause potentially different issues if they weren't all canonical.

Basically, canonicalizing, or even checking when it's a no-op is quite expensive if we know they are in the right format.

Reply:
You may choose to encode errors without canonicalization as we export the header field on [`Meta()`](https://pkg.go.dev/connectrpc.com/connect#Error.Meta). So:
```
connectErr.Meta()["content-length"] = []string{"1234"}
```
Would cause issues. This was done as a precaution but it could be removed.

Reply:
Ah, ok, just making sure this was potentially from user input. I think it's fine given it's only on errors anyways.